File: Python3/804. Unique Morse Code Words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        codes = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        
        ans = set()
        a = ord('a')
        
        for w in words:
            trans = ""
            for c in w:
                trans += codes[ord(c) - a]
            ans.add(trans)
            
        return len(ans)

File: Java/804. Unique Morse Code Words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] codes = new String[]{".-","-...","-.-.","-..",".","..-.","--.",
                         "....","..",".---","-.-",".-..","--","-.",
                         "---",".--.","--.-",".-.","...","-","..-",
                         "...-",".--","-..-","-.--","--.."};
        
        Set<String> ans = new HashSet();
        
        for (String w: words) {
            StringBuilder sb = new StringBuilder();
            for (char c: w.toCharArray()) {
                sb.append(codes[c - 'a']);
            }
            ans.add(sb.toString());
        }
        
        return ans.size();
    }
}